server/src/main/java/io/druid/server/coordinator/HttpLoadQueuePeon.java

```diff
@@ -0,0 +1,603 @@
+                            log.error("WTF! Server[%s] returned unknown state in status[%s].", serverId, e.rhs);
+                        }
+                      }
+                    }
+                  }
+                  catch (Exception ex) {
+                    scheduleNextRunImmediately = false;
+                    onFailure(ex);
+                  }
+                  finally {
+                    mainLoopInProgress.set(false);
+                  }
+                } else {
+                  scheduleNextRunImmediately = false;
+                  onFailure(new RE("Unexpected Response Status."));
+                }
+
+                if (scheduleNextRunImmediately) {
+                  processingExecutor.execute(HttpLoadQueuePeon.this::doSegmentManagement);
+                }
+              }
+
+              @Override
+              public void onFailure(Throwable t)
+              {
+                try {
+                  log.info(
+                      "Request[%s] Failed with code[%s] and status[%s]. Reason[%s]. Error [%s]",
+                      changeRequestURL,
+                      responseHandler.status,
+                      responseHandler.description,
+                      t.getMessage()
+                  );
+                  log.error(
+                      t,
+                      "Request[%s] Failed with code[%s] and status[%s]. Reason[%s].",
+                      changeRequestURL,
+                      responseHandler.status,
+                      responseHandler.description
+                  );
+                }
+                finally {
+                  mainLoopInProgress.set(false);
+                }
+              }
+            },
+            processingExecutor
+        );
+      }
+      catch (Throwable th) {
+        mainLoopInProgress.set(false);
+      }
+    } else {
+      log.debug("[%s] No segments to load/drop.");
+      mainLoopInProgress.set(false);
+    }
+  }
+
+  @Override
+  public void start()
+  {
+    synchronized (lock) {
+      if (stopped) {
+        throw new ISE("Can't start.");
+      }
+
+      ScheduledExecutors.scheduleAtFixedRate(
+          processingExecutor,
+          new Duration(config.getHttpLoadQueuePeonRepeatDelay()),
+          new Callable<ScheduledExecutors.Signal>()
+          {
+            @Override
+            public ScheduledExecutors.Signal call()
+            {
+              if (!stopped) {
+                doSegmentManagement();
+              }
+
+              if (stopped) {
+                return ScheduledExecutors.Signal.STOP;
+              } else {
+                return ScheduledExecutors.Signal.REPEAT;
+              }
+            }
+          }
+      );
+    }
+  }
+
+  @Override
+  public void stop()
+  {
+    synchronized (lock) {
+      if (stopped) {
+        return;
+      }
+
+      stopped = true;
+
+      for (SegmentHolder holder : segmentsToDrop.values()) {
+        holder.requestCompleted(false, null);
```

Comment:
This is acting like these requests succeeded, is that the correct thing to do?

Reply:
yes, this behavior is retained from existing behavior in CuratorLoadQueuePeon (copy of older LoadQueuePeon class )